### ISSUE TITLE
Unify Geometry::inputVertexPositions and Geometry::vertexPositions

### ISSUE THREAD

**Issue by keenancrane** [nmwsharp/geometry-central#88]
It violates the principle of least surprise for a `Geometry` object to have two different notions of position (`inputVertexPositions` and `vertexPositions`).  The paradigm in GeometryCentral is that each `Geometry` object represents a single embedding of the surface.  If someone wants to preserve the input geometry, they should just make a copy of the input `Geometry`.

This redundancy is especially confusing for methods like `WriteSurfaceMesh`, where it's not clear (without reading the source...) which positions will be used.

The proposal is to eliminate `inputVertexPositions` and keep just `vertexPositions`.  This will break all sorts of things / existing code, so perhaps we can just deprecate `inputVertexPositions` for a version or two.

**Comment by keenancrane** [comment]
At bare minimum, if anything in Geometry Central is going to depend on `Geometry::vertexPositions` by default (as mesh I/O currently is), seems we should really allocate `Geometry::vertexPositions` by default.

For instance, this morning I was segfaulting because I didn't realize I needed to do something like

```geometry->vertexPositions = geometry->inputVertexPositions;```

Otherwise `vertexPositions` is not allocated.

Note that the documentation currently says:

> In the common case of a VertexPositionGeometry, this member is a copy of the input vertex positions,

which seems not to be true in the latest version.

**Comment by nmwsharp** [comment]
Quick note: for the particular case mentioned above, we need to call `geometry->requireVertexPositions()` similar to other quantities, and it will populate the necessary fields (for `VertexPositionGeometry`, this internally does the copy you mention).

---

More generally, the distinction between `vertexPositions` and `inputVertexPositions` has definitely been a source of confusion. We originally distinguished these to nicely support hypothetical geometries where vertex positions themselves are derived data. However, these haven't really materialized, and I agree that it's more important that the `VertexPositionGeometry` be as simple as possible, since it's the overwhelming common case, especially for first time users. I had been holding off doing something about this to group a bunch of changes in to a breaking "Version 2" release, but instead I think a small change will suffice here:

I just pushed commits 13208f02196c095f0f97c24ec81ec400fcef9b9b and 1af6a29d427413feee3689a51c6ff1681eee893b which alias `vertexPositions` and `inputVertexPositions` to point to the same underlying array. The function  `requireVertexPositions()` still exists, but is a no-op for the `VertexPositionGeometry`. Both `vertexPositions` and `inputVertexPositions` are still there, and existing code that read/writes to them will still work fine. Teeeeechnically even this change can break existing code, but only if a user is intentionally storing different data in `inputVertexPositions` and `vertexPositions`, which seems pretty unusual, and goes against the recommended usage. I think this change is safe enough to be OK to push straight to the main branch. (All the same is also done likewise with `inputEdgeLengths` for the `EdgeLengthGeometry`).

With this change, code can just always work with `geometry->vertexPositions` and everything will behave as one would probably expect. I also went through and updated the docs and tutorial to reflect this. I think that this should be a pretty good quick-fix in lieu of a deep rethink of the design.
